operator/halo2kzg/lib/src/lib.rs

```diff
@@ -0,0 +1,398 @@
+use std::{
+    fs::File,
+    io::{BufReader, BufWriter, ErrorKind, Write, Read},
+};
+use ff::{Field, PrimeField};
+use halo2_proofs::{
+    circuit::{Layouter, SimpleFloorPlanner, Value},
+    plonk::{
+        create_proof, keygen_pk, keygen_vk_custom, pk_read, vk_read, verify_proof, Advice, Circuit, Column,
+        ConstraintSystem, ErrorFront, Fixed, Instance,
+    },
+    poly::{
+        kzg::{
+            commitment::{KZGCommitmentScheme, ParamsKZG},
+            multiopen::{ProverSHPLONK, VerifierSHPLONK},
+            strategy::SingleStrategy,
+        },
+        commitment::Params,
+        Rotation,
+    },
+    transcript::{
+        Blake2bRead, Blake2bWrite, Challenge255, TranscriptReadBuffer, TranscriptWriterBuffer,
+    },
+    SerdeFormat,
+};
+use halo2curves::bn256::{Bn256, Fr, G1Affine};
+use rand_core::OsRng;
+
+pub const MAX_PROOF_SIZE: usize = 2048;
+
+//TODO: write this out based on size of G1Affine elements
+pub const MAX_KZG_PARAMS_SIZE: usize = 2308;
+
+//TODO: write this out based on size of G1Affine elements
+pub const MAX_VERIFIER_KEY_SIZE: usize = 518;
+
+//TODO: write this out based on size of G1Affine elements
+pub const MAX_PUBLIC_INPUT_SIZE: usize = 64;
+
+//NOTE(pat): For now we only support a single strategy and single parameter configuration.
+//NOTE(pat): We can't use generics over FFI so we need to have multiple implementations for each field/curve -> Use Bn254
+//NOTE(pat): We can't use generics over FFI so we need multiple implementations for each plonk implementation
```

Comment:
Polymorphism may be usable in FFI with some effort, as it would be pointer passing mostly. I could check if you want.

Reply:
We aim to initially support Bn254 Field elements so for now this is not a priority.